azure-keyvault/azure/keyvault/keys/_models.py

```diff
@@ -1,406 +1,216 @@
-        self.managed = None
+    @property
+    def tags(self):
+        # type: () -> Dict[str, str]
+        """Application specific metadata in the form of key-value pairs.
+        :rtype: dict"""
+        return self._tags
 
 
-class DeletedKeyItem(KeyItem):
-    """The deleted key item containing the deleted key metadata and information
-    about deletion.
-    Variables are only populated by the server, and will be ignored when
-    sending a request.
-    :param kid: Key identifier.
-    :type kid: str
-    :param attributes: The key management attributes.
-    :type attributes: ~azure.keyvault.v7_0.models.KeyAttributes
-    :param tags: Application specific metadata in the form of key-value pairs.
-    :type tags: dict[str, str]
-    :ivar managed: True if the key's lifetime is managed by key vault. If this
-     is a key backing a certificate, then managed will be true.
-    :vartype managed: bool
-    :param recovery_id: The url of the recovery object, used to identify and
-     recover the deleted key.
-    :type recovery_id: str
-    :ivar scheduled_purge_date: The time when the key is scheduled to be
-     purged, in UTC
-    :vartype scheduled_purge_date: datetime
-    :ivar deleted_date: The time when the key was deleted, in UTC
-    :vartype deleted_date: datetime
+class Key(KeyBase):
+    """A key consisting of all KeyBase and key_material information.
     """
 
-    _validation = {
-        "managed": {"readonly": True},
-        "scheduled_purge_date": {"readonly": True},
-        "deleted_date": {"readonly": True},
-    }
+    def __init__(self, attributes, key, **kwargs):
+        super(Key, self).__init__(attributes, key.kid, **kwargs)
+        self._key_material = key
+
+    @classmethod
+    def _from_key_bundle(cls, key_bundle):
+        # type: (models.KeyBundle) -> Key
+        """Construct a key from an autorest-generated KeyBundle"""
+        return cls(
+            key_bundle.attributes,
+            key_bundle.key,
+            managed=key_bundle.managed,
+            tags=key_bundle.tags,
+        )
 
-    _attribute_map = {
-        "kid": {"key": "kid", "type": "str"},
-        "attributes": {"key": "attributes", "type": "KeyAttributes"},
-        "tags": {"key": "tags", "type": "{str}"},
-        "managed": {"key": "managed", "type": "bool"},
-        "recovery_id": {"key": "recoveryId", "type": "str"},
-        "scheduled_purge_date": {"key": "scheduledPurgeDate", "type": "unix-time"},
-        "deleted_date": {"key": "deletedDate", "type": "unix-time"},
-    }
-
-    def __init__(self, **kwargs):
-        super(DeletedKeyItem, self).__init__(**kwargs)
-        self.recovery_id = kwargs.get("recovery_id", None)
-        self.scheduled_purge_date = None
-        self.deleted_date = None
+    @property
+    def key_material(self):
+        # type: () -> Dict[str, str]
+        return self._key_material
 
 
-class DeletedKeyItemPaged(Paged):
+class DeletedKey(KeyBase):
```

Comment:
Is `DeletedKey` simply `Key` plus `deleted_date`, `recovery_id`, and `scheduled_purge_date`? Then it can subclass `Key` and add those properties.